packages/google-vertex/src/google-vertex-image-model.ts

```diff
@@ -84,10 +88,22 @@ export class GoogleVertexImageModel implements ImageModelV1 {
       fetch: this.config.fetch,
     });
 
+    if (!response.predictions) {
+      const response: ImageModelResponseMetadata = {
+        timestamp: currentDate,
+        modelId: this.modelId,
+        headers: responseHeaders,
+      };
+      throw new NoImageGeneratedError({
+        response,
+      });
+    }
+
     return {
-      images: response.predictions.map(
-        (p: { bytesBase64Encoded: string }) => p.bytesBase64Encoded,
-      ),
+      images:
+        response.predictions.map(
+          (p: { bytesBase64Encoded: string }) => p.bytesBase64Encoded,
+        ) ?? [],
```

Comment:
I propose moving the error throwing logic into `generateImage`. It has the following advantages:

- works for all image providers without any extra provider code
- no dependency on the `ai` package